src/features/defi/contexts/IdleProvider/IdleProvider.tsx

```diff
@@ -0,0 +1,54 @@
+import type { ChainAdapter } from '@shapeshiftoss/chain-adapters'
+import { IdleInvestor } from '@shapeshiftoss/investor-idle'
+import { KnownChainIds } from '@shapeshiftoss/types'
+import { getConfig } from 'config'
+import React, { PropsWithChildren, useContext, useEffect, useState } from 'react'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { selectFeatureFlags } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+type IdleContextProps = {
+  loading: boolean
+  enabled: boolean
+  idle: IdleInvestor | null
+}
+
+const IdleContext = React.createContext<IdleContextProps | null>(null)
+
+export const useIdle = () => {
+  const context = useContext(IdleContext)
+  if (!context) throw new Error("useIdle can't be used outside of the IdleProvider")
+  return context
+}
+
+export const IdleProvider: React.FC<PropsWithChildren> = ({ children }) => {
+  const [idle, setIdle] = useState<IdleInvestor | null>(null)
```

Comment:
This should probably be `idleInvestor` vs `idle`. At consumption time, this isn't clear what `idle` is.